scm/src/main/java/com/sandboni/core/scm/resolvers/CliChangeScopeResolver.java

```diff
@@ -0,0 +1,154 @@
+        super(repository, filters);
+    }
+
+    private static <T> Consumer<T> throwingConsumerWrapper(ThrowingConsumer<T, Exception> throwingConsumer) {
+        return i -> {
+            try {
+                throwingConsumer.accept(i);
+            } catch (Exception ex) {
+                throw new SourceControlRuntimeException("Unable to process diff", ex);
+            }
+        };
+    }
+
+    @Override
+    public ChangeScope<Change> getChangeScope(RevisionScope<ObjectId> revisionScope) throws SourceControlException {
+        final ChangeScope<Change> scope = new ChangeScopeImpl();
+
+        List<Diff> diffs = GitDiffRunner.diff(repository, revisionScope);
+
+        ObjectId toRevision = revisionScope.getTo();
+
+        diffs.forEach(throwingConsumerWrapper(d -> getChangesForDiff(d, scope, toRevision)));
+
+        return scope;
+    }
+
+
+    private void getChangesForDiff(Diff diff, ChangeScope<Change> scope, ObjectId toRevision) throws IOException {
+
+        String fromFileName = trim(diff.getFromFileName());
+        String toFileName = trim(diff.getToFileName());
+
+        log.debug("processing diff: from file: {} ,to file {}", fromFileName, toFileName);
+
+        if (fileDeleted(toFileName)) {
+            return;
+        }
+
+        String content = getFileContent(toRevision, toFileName);
+        String fqnFileName = RawUtil.getFullClassPath(content, toFileName);
+
+        diff.getHunks().forEach(h -> processChanges(getChangesFromHunk(h), fqnFileName, content).forEach(scope::addChange));
+    }
+
+
+    private String getFileContent(ObjectId revision, String fileName) throws IOException {
+        return rawTextToString.apply(getRawText(revision, fileName));
+    }
+
+    private String trim(String fromFileName) {
+        return trimFileName.apply(fromFileName);
+    }
+
+    private boolean fileDeleted(String toFileName) {
+        return toFileName.equals(NON_EXISTING_FILE);
+    }
+
+    private Stream<Change> processChanges(Map<ChangeType, Set<Integer>> changes, String filePath, String content) {
+        return changes.entrySet().parallelStream().map(e -> new SCMChangeBuilder().with(scm -> {
+            scm.path = filePath;
+            scm.changedLines = e.getValue();
+            scm.changeType = e.getKey();
+            scm.fileContent = content;
+            scm.repository = repository.getWorkTree().getAbsolutePath();
+        }).build());
+    }
+
+    private Map<ChangeType, Set<Integer>> getChangesFromHunk(Hunk hunk) {
+        Range fromFileRange = hunk.getFromFileRange();
+        int currentLine = Math.max(fromFileRange.getLineStart(), 1); // for new files, hunk line will start with 0
+        List<Integer> pendingLines = new LinkedList<>();
+        Map<ChangeType, Set<Integer>> changes = new HashMap<>();
+
+        List<Line> lines = hunk.getLines();
+        for (Line line : lines) {
+            switch (line.getLineType()) {
+                case NEUTRAL:
```

Comment:
are those all available cases? if not, do we want to add a `default` ?

Reply:
These are all possible cases. there will be no others (see UnifiedDiffParser)